source/braille.py

```diff
@@ -3841,3 +3844,20 @@ def _speakOnRouting(info: textInfos.TextInfo):
 
 	info.expand(textInfos.UNIT_CHARACTER)
 	spellTextInfo(info)
+
+
+def _speakOnNavigatingByUnit(info: textInfos.TextInfo, readingUnit: str) -> None:
+	"""Speaks the reading unit after navigating by it with braille.
+
+	:param info: The TextInfo at the cursor position after navigating.
+	:param readingUnit: The reading unit to expand TextInfo.
+	"""
+	if not config.conf["braille"]["speakOnNavigatingByUnit"]:
+		return
+	# Import late to avoid circular import.
+	from speech.speech import cancelSpeech, speakTextInfo
+
+	cancelSpeech()
+	copy = info.copy()
+	copy.expand(readingUnit)
+	speakTextInfo(copy, unit=readingUnit, reason=controlTypes.OutputReason.CARET)
```

Comment:
Is there a reason to `cancelSpeech` before copying and expanding the `TextInfo`, rather than immediately before `speakTextInfo`?
```suggestion
	copy = info.copy()
	copy.expand(readingUnit)
	cancelSpeech()
	speakTextInfo(copy, unit=readingUnit, reason=controlTypes.OutputReason.CARET)
```